### PR TITLE
feat: remove walletConnect v1 and use v2 by default

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Context: https://cowservices.slack.com/archives/C0375NV72SC/p1695724802482379

Looks like V2 is already well-supported and we even started receiving warnings about using V1.
So, it's time to remove V1 and use V2 by default.

  # To Test

1. There is no V1 in the wallets list
2. V2 is available without feature-flag
3. Alpha, Ambire and Trust icons are using V2

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/wallet/containers/WalletModal/index.tsx [hunk 1]
```diff
@@ -90,21 +90,6 @@ export function WalletModal() {
 
         await connector.activate(getCurrentChainIdFromUrl())
 
-        const connection = getWeb3ReactConnection(connector)
-
-        // Important for balances to load when connected to Gnosis-chain via WalletConnect
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/wallet/containers/WalletModal/index.tsx:95]: Not needed at all anymore?

File: apps/cowswap-frontend/src/modules/wallet/pure/StatusIcon/index.tsx [hunk 2]
```diff
@@ -28,9 +28,6 @@ export function StatusIcon({ connectionType }: StatusIconProps) {
     case ConnectionType.INJECTED:
       image = <Identicon />
       break
-    case ConnectionType.WALLET_CONNECT:
-      image = <img src={WalletConnectIcon} alt="WalletConnect" />
-      break
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/wallet/pure/StatusIcon/index.tsx:33]: Doesn't v2 have an icon?

File: apps/cowswap-frontend/src/modules/wallet/containers/WalletModal/index.tsx (hunk 1 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/wallet/containers/WalletModal/index.tsx:95]: Nope, it was a fix only for V1

File: apps/cowswap-frontend/src/modules/wallet/pure/StatusIcon/index.tsx (hunk 2 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/wallet/pure/StatusIcon/index.tsx]: It didn't before, but I think you are right, it should have. Added

**Review by alfetopito — APPROVED** [review]
One last nitpick.

Since we do not have another option, might as well remove the tooltip?

image

**Comment by shoom3301** [comment]
@alfetopito thanks! Removed the tooltip

**Pull request merged**